### PR TITLE
Add support for structural types

### PR DESCRIPTION
**Pull request by LukasKellenberger**

### CODE REVIEW

**Comment by densh** [comment]
Needs a rebase & reformat due to scalafmt upgrade.

**Comment by densh** [comment]
@LukasKellenberger Please rebase your PR on top of latest master.

File: nir/src/main/scala/scala/scalanative/nir/Versions.scala [hunk 1]
```diff
@@ -21,7 +21,7 @@ object Versions {
    * new version of the toolchain.
    */
   final val compat: Int   = 8
-  final val revision: Int = 10
+  final val revision: Int = 9
```
Comment (densh) [inline nir/src/main/scala/scala/scalanative/nir/Versions.scala]: Decreasing number is probably an artifact of rebases. Has to be `compat = 8, revision = 11`

Comment (LukasKellenberger) [inline nir/src/main/scala/scala/scalanative/nir/Versions.scala]: 🙈 It's fixed now.

**Comment by densh** [comment]
LGTM, fantastic work @LukasKellenberger !

**Pull request merged**